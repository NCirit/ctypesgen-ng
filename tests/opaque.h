struct handle;

typedef struct handle handle_t;

struct handle {
	char *name;
};

struct opaque;

typedef struct opaque *opaque_t;

opaque_t opaque_new(void);

void opaque_free(opaque_t opaque);
