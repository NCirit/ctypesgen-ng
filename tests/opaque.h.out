### This file automatically created by ctypesgen-ng, the commandline parameters passed to it are given below
### /app/ctypesgen.py --clang-flags "-Wno-error=implicit-int" --ignore-included --library test.so --headers tests/opaque.h

import ctypes
import ctypes.util
import os


if not ('LIBRARY_PATH' in os.environ):
    if 'PYTHONPATH' in os.environ:
        os.environ['LIBRARY_PATH'] = os.environ['PYTHONPATH']
    else:
        os.environ['LIBRARY_PATH'] = os.path.dirname(os.path.abspath(__file__))

if os.name == "nt":
    if 'PYTHONPATH' in os.environ:
        os.environ['PATH'] += os.pathsep + os.environ['PYTHONPATH']
    else:
        os.environ['PATH'] += os.pathsep + os.path.dirname(os.path.abspath(__file__))
test_path = ctypes.util.find_library('test')
test = ctypes.CDLL(os.path.abspath(test_path))


class struct_handle(ctypes.Structure):
    pass

typedef_handle_t = struct_handle

struct_handle._pack_ = 1
struct_handle._fields_ = [
    ('name', ctypes.c_char_p),
]


class struct_opaque(ctypes.Structure):
    pass

typedef_opaque_t = ctypes.POINTER(struct_opaque)

test.opaque_new.restype = typedef_opaque_t
test.opaque_new.argtypes = []

test.opaque_free.restype = None
test.opaque_free.argtypes = [typedef_opaque_t]
